### PR TITLE
Add debug option for infinispan device registry

### PR DESCRIPTION
**Pull request by jbtrystram**
### Type of change

- Bugfix
- Enhancement / new feature
- Refactoring
- Documentation

### Description

Enable debug output in the infinispan device registry with :
```
kind: IoTConfig
spec:
  services:
    deviceRegistry:
      infinispan:
        java:
          debug: true
```

### CODE REVIEW

**Comment by jbtrystram** [comment]
@enmasse-ci run tests profile=iot

File: pkg/controller/iotconfig/utils.go [hunk 1]
```diff
@@ -62,12 +62,21 @@ func AppendEnvVarValue(container *corev1.Container, name string, value string) {
...
+	if debug {
+		opts += " -Ddebug"
+
+	}
 	AppendEnvVarValue(
-		container,
-		"JAVA_APP_OPTS",
-		"-Djava.net.preferIPv4Stack=true -Dvertx.logger-delegate-factory-class-name=io.vertx.core.logging.SLF4JLogDelegateFactory",
-	)
+			container,
```
Comment (ctron) [inline pkg/controller/iotconfig/utils.go]: I think you need to run `go fmt` against the changed files.

**Comment by ctron** [comment]
I am struggling a bit with this PR. I think it is implemented well! IIRC we would only also need an additional port mapping.

The question is just, should it be that simple to enable this internal functionality. Having that enabled, would allow you direct access, with not authentication, to the credentials database. Which I would say is something that shouldn't be enabled that easily.

**Comment by jbtrystram** [comment]
@ctron I agree with you, the original idea was to enable debug output to facilitate tests debugging ! The controller should alter the logback config file, but I wanted to see if that worked with an extra java argument.. And then I looked at the logs and realized that this was not the effect I expected :) 
That why there is no port mapping

**Comment by jbtrystram** [comment]
@enmasse-ci run tests profile=iot testcase=io.enmasse.systemtest.iot.shared.http.HttpAdapterTest

**Comment by jbtrystram** [comment]
@enmasse-ci run tests profile=iot

**Pull request closed without merging**